casper/src/test/scala/coop/rchain/casper/HashSetCasperTest.scala

```diff
@@ -523,6 +536,79 @@ class HashSetCasperTest extends FlatSpec with Matchers {
     node.tearDown()
   }
 
+  it should "allow paying for deploys" in {
+    val node      = HashSetCasperTestNode.standalone(genesis, validatorKeys.head)
+    val (sk, pk)  = Ed25519.newKeyPair
+    val user      = ByteString.copyFrom(pk)
+    val timestamp = System.currentTimeMillis()
+    val phloPrice = 1L
+    val amount    = 847L
+    val sigDeployData = ProtoUtil
+      .sourceDeploy(
+        s"""new retCh in { @"blake2b256Hash"!([0, $amount, *retCh].toByteArray(), "__SCALA__") }""",
+        timestamp,
+        accounting.MAX_VALUE
+      )
+      .withUser(user)
+    val sigData = node.runtimeManager
+      .captureResults(
+        genesis.getBody.getPostState.tuplespace,
+        ProtoUtil.deployDataToDeploy(sigDeployData)
+      )
+      .head
+      .exprs
+      .head
+      .getGByteArray
```

Comment:
OK, again - I see it now. You rely on that the name generated will be the one you use as a destination channel for transfer.

Reply:
Yes, I am relying on being able to consistently generate the same name. Which does not preclude doing it in Scala because the logic for choosing the name is in the interpreter code somewhere, but lacking a complete understanding of how that works I chose to do it this way.

Reply:
Currently available alternative is to write the rholang AST in Scala. That's what I did but it doesn't scale well and  is error prone.